src/openvpn/ps.c

```diff
@@ -425,7 +433,11 @@ proxy_entry_new(struct proxy_connection **list,
     struct proxy_connection *cp;
 
     /* connect to port share server */
+#if defined(TARGET_LINUX) && defined(ENABLE_MPTCP)   
+    if ((sd_server = socket(PF_INET, SOCK_STREAM, IPPROTO_MPTCP)) < 0)
```

Comment:
> here you might want to avoid duplicating everything and do something like:
> 
> ```
> if ((sd_server = socket(PF_INET, SOCK_STREAM,
>                         options->enable_mptcp ? IPPROTO_MPTCP, IPPROTO_TCP)) < 0)
> ```

Reply:
@matttbe I don't know OpenVPN at all so I'm trying to figure out how I would pass the options structure to the function where this code runs.